### DISCUSSION TITLE
Bazel packaging system

### DISCUSSION THREAD

**Discussion by dmadisetti** [bazel-contrib/SIG-rules-authors#18] [General]
Between standardizing rules (https://github.com/bazel-contrib/SIG-rules-authors/issues/6) and attaching ownership/ metadata to rules (https://github.com/bazel-contrib/SIG-rules-authors/issues/5)

Wouldn't it just make sense to come up with some sort of packaging system, like an AUR, NPM, cargo or PIP?

Opening this as a discussion question.

**Comment by alexeagle** [comment]
I think this is already well-established as .tgz files containing the workspace content, with bazel-central-registry to host the metadata. Most rulesets ship this way and AFAIK that is working well.

What would you want some other packaging to consist of? What use case could it solve?

**Comment by aherrmann** [comment]
If I understand the question correctly, then the [upcoming `bzlmod`](https://youtu.be/TxOCKtU39Fs) would provide this packaging system.

**Comment by cgrindel** [comment]
Now that bzlmod is released. I think that we should unpin this discussion. Thoughts?

**Comment by aherrmann** [comment]
As I understand the request, I think [bzlmod](https://bazel.build/external/overview#bzlmod) and the [BCR](https://github.com/bazelbuild/bazel-central-registry) (or potentially other registries) do already provide this: The BCR can be seen as the main package repository. Anyone who would like to contribute a Bazel module, can open a PR to add it there. The standard package format is, as @alexeagle points out above, just a tarball containing the relevant files. Maybe I'm missing something about the question though. @dmadisetti did you have something in mind that is not served by bzlmod and the BCR (or other registries)?

**Reply by hobofan** [reply]
> Maybe I'm missing something about the question though

I think you may be missing the timestamp of when that question was originally posed 😉  (pre-bzlmod times)

**Reply by aherrmann** [reply]
I'm aware, I had linked a presentation about bzlmod to that end in [my earlier reply](https://github.com/bazel-contrib/SIG-rules-authors/discussions/18#discussioncomment-1686717). This message was intended in response to https://github.com/bazel-contrib/SIG-rules-authors/discussions/18#discussioncomment-6346328.
